packages/zero-server/src/adapters/drizzle-pg.ts

```diff
@@ -1,74 +1,152 @@
-import type {NodePgClient, NodePgDatabase} from 'drizzle-orm/node-postgres';
+import type {
+  NodePgClient,
+  NodePgDatabase,
+  NodePgQueryResultHKT,
+} from 'drizzle-orm/node-postgres';
+import type {PgTransaction} from 'drizzle-orm/pg-core';
+import type {ExtractTablesWithRelations} from 'drizzle-orm/relations';
+import type {Row} from '../../../zero-protocol/src/data.ts';
 import type {
   DBConnection,
   DBTransaction,
 } from '../../../zql/src/mutate/custom.ts';
-import type {Row} from '../../../zero-protocol/src/data.ts';
-import type {Schema} from '../../../zero-schema/src/builder/schema-builder.ts';
 import {ZQLDatabase} from '../zql-database.ts';
+import type {Schema} from '../../../zero-schema/src/builder/schema-builder.ts';
 
-type DrizzleConnection<T extends Record<string, unknown>> =
-  NodePgDatabase<T> & {
+/**
+ * Helper type for the wrapped transaction used by drizzle-orm/node-postgres.
+ *
+ * @remarks Use with `ServerTransaction` as `ServerTransaction<Schema, NodePgDrizzleTransaction>`.
+ */
+export type NodePgDrizzleTransaction<
```

Comment:
It seems wrong to me that there is both a PG Drizzle adapter and also a Postgres Drizzle adapter. We should not have a combinatorial explosion here, but instead the postgres and drizzle adapters should be composable.I think maybe the Drizzle adapter should be able to take any underlying db adapter and expose it or something.

Reply:
Yeah I agree with you. I took a stab at this and the types get very messy trying to combine them all into one Drizzle wrapper. It got so convoluted that I reverted it. But I changed the Drizzle adapters reuse more code from the base adapters.

FWIW Drizzle doesn't support many more PG adapters outside of these two. And most of this PR is just typing helpers so they get a nice `wrappedTransaction` type.

Reply:
Does Drizzle expose a higher level abstraction for running SQL that doesn't require hooking into the underlying pg/postgres?

my 2 cents:
1. we should support `pg` and `postgres` for drizzle only
2. if someone has something more exotic they use with drizzle, they can write the adapter themselves at that point

Reply:
Thanks for this - I just reworked to merge the two drizzle adapters into one. So now any drizzle client can be used with:

```tsx
const drizzleDb = drizzle(pool, { schema: drizzleSchema });
const zql = zeroDrizzle(schema, drizzleDb);
```

This also supports:

```tsx
import { type DrizzleTransaction } from '@rocicorp/zero/server/adapters/drizzle';
import type { ServerTransaction } from '@rocicorp/zero';

type ServerTx = ServerTransaction<
  Schema,
  DrizzleTransaction<typeof drizzleDb>
>;

// then
await tx.dbTransaction.wrappedTransaction
    .insert(drizzleSchema.user)
    .values({ id, name });
```